### PR TITLE
msg/async: allow connection reaping to be tuned; fix cephfs test

### PR DESCRIPTION
**Pull request by liewegas**
Fixes: https://tracker.ceph.com/issues/50622

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephfs/test_sessionmap.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def test_mount_conn_close(self):
         That when a client unmounts, the thread count on the MDS goes back
         to what it was before the client mounted
         """
+        self.run_cluster_cmd(['config', 'set', 'mds', 'ms_async_reap_threshold', '1'])
```
Comment (batrick) [inline qa/tasks/cephfs/test_sessionmap.py:59]:
```suggestion
        self.config_set('mds', 'ms_async_reap_threshold', '1')
```

**Review by batrick — CHANGES_REQUESTED** [review]

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
Thanks for fixing this @liewegas !

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-05-21_12:49:52-rados-wip-kefu-testing-2021-05-21-1851-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-05-22_15:42:09-rados-wip-kefu-testing-2021-05-22-2207-distro-basic-smithi/

failure tracked by 

- https://tracker.ceph.com/issues/50243
- https://tracker.ceph.com/issues/50151

**Review by batrick — CHANGES_REQUESTED** [review]
`test_tell_conn_close` needs updated too.

See: /ceph/teuthology-archive/pdonnell-2021-06-03_03:40:33-fs-wip-pdonnell-testing-20210603.020013-distro-basic-smithi/6148218/teuthology.log

Also, I found that reaping does not happen immediately. I had to make a change suggested below to get it right.

````
diff --git a/qa/tasks/cephfs/test_sessionmap.py b/qa/tasks/cephfs/test_sessionmap.py
index c600811bccc..ad6fd1d609c 100644
--- a/qa/tasks/cephfs/test_sessionmap.py
+++ b/qa/tasks/cephfs/test_sessionmap.py
@@ -41,15 +41,18 @@ class TestSessionMap(CephFSTestCase):
         the conn count goes back to where it started (i.e. we aren't
         leaving connections open)
         """
+        self.config_set('mds', 'ms_async_reap_threshold', '1')
+
         self.mount_a.umount_wait()
         self.mount_b.umount_wait()
 
         status = self.fs.status()
         s = self._get_connection_count(status=status)
         self.fs.rank_tell(["session", "ls"], status=status)
-        e = self._get_connection_count(status=status)
-
-        self.assertEqual(s, e)
+        self.wait_until_true(
+            lambda: self._get_connection_count(status=status) == s,
+            timeout=30
+        )
 
     def test_mount_conn_close(self):
         """
@@ -66,9 +69,10 @@ class TestSessionMap(CephFSTestCase):
         self.mount_a.mount_wait()
         self.assertGreater(self._get_connection_count(status=status), s)
         self.mount_a.umount_wait()
-        e = self._get_connection_count(status=status)
-
-        self.assertEqual(s, e)
+        self.wait_until_true(
+            lambda: self._get_connection_count(status=status) == s,
+            timeout=30
+        )
 
     def test_version_splitting(self):
         """
````

**Comment by liewegas** [comment]
Thanks, updated!

**Review by batrick — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-06-13_08:30:20-rados-wip-kefu-testing-2021-06-13-1215-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-06-13_13:23:30-rados-wip-kefu-testing-2021-06-13-2001-distro-basic-smithi/

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2021-June-13

**Pull request merged**